packages/react/src/components/List/HierarchyList/HierarchyList.jsx

```diff
@@ -40,7 +41,7 @@ const propTypes = {
   /** Buttons to be presented in List header */
   buttons: PropTypes.arrayOf(PropTypes.node),
   /** ListItems to be displayed */
-  items: PropTypes.arrayOf(PropTypes.any).isRequired,
+  items: PropTypes.arrayOf(PropTypes.shape(ListItemPropTypes)),
```

Comment:
This is a breaking change of an item that we just took out of experimental. We should make this an enum with the deprecation warning for the old way included. I see it's no longer required so what behavior is changed by this new shape?

Reply:
This enforces the same prop types as the List below expects so the `any` was always wrong and would throw prop types warnings regardless

Reply:
Yeah I see down in the list now. Thanks.